webpack.config.js

```diff
@@ -1,36 +1,48 @@
-const path = require('path');
+const path = require("path");
 
 // Base config that applies to either development or production mode.
 const config = {
-  entry: './src/index.js',
+  entry: "./src/index.ts",
   output: {
-    library: 'ScratchBlocks',
-    libraryTarget: 'commonjs2',
-    path: path.resolve(__dirname, 'dist'),
-    filename: '[name].js'
+    library: "ScratchBlocks",
+    libraryTarget: "commonjs2",
+    path: path.resolve(__dirname, "dist"),
+    filename: "[name].js",
+  },
+  resolve: {
+    extensions: [".ts", ".js"],
+  },
+  module: {
+    rules: [
+      {
+        test: /\.ts$/,
+        use: "ts-loader",
+        exclude: /node_modules/,
+      },
+    ],
   },
   // Enable webpack-dev-server to get hot refresh of the app.
   devServer: {
-    static: './build',
+    static: "./build",
   },
 };
 
 module.exports = (env, argv) => {
-  if (argv.mode === 'development') {
+  if (argv.mode === "development") {
     // Set the output path to the `build` directory
     // so we don't clobber production builds.
-    config.output.path = path.resolve(__dirname, 'build');
+    config.output.path = path.resolve(__dirname, "build");
 
     // Generate source maps for our code for easier debugging.
     // Not suitable for production builds. If you want source maps in
     // production, choose a different one from https://webpack.js.org/configuration/devtool
-    config.devtool = 'eval-cheap-module-source-map';
+    config.devtool = "eval-cheap-module-source-map";
```

Comment:
Create an issue to track either removing or optimizing source maps in the future, and add a TODO here.

Reply:
Added a comment to https://github.com/gonfunko/scratch-blocks/issues/15 to realign the webpack configs.